sensor/tests/resource/helper.go

```diff
@@ -324,6 +325,37 @@ func (c *TestContext) LastViolationStateWithTimeout(name string, assertion Alert
 
 }
 
+// LastViolationStateByID checks the violation state by deployment ID
+func (c *TestContext) LastViolationStateByID(id string, assertion AlertAssertFunc, message string, checkEmptyAlertResults bool) {
+	c.LastViolationStateByIDWithTimeout(id, assertion, message, checkEmptyAlertResults, 3*time.Second)
+}
+
+// LastViolationStateByIDWithTimeout checks that a violation state for a deployment must match `assertion`. If violation state does not match
+// until `timeout` the test fails.
+func (c *TestContext) LastViolationStateByIDWithTimeout(id string, assertion AlertAssertFunc, message string, checkEmptyAlertResults bool, timeout time.Duration) {
+	timer := time.NewTimer(timeout)
+	ticker := time.NewTicker(10 * time.Millisecond)
+	var lastErr error
```

Comment:
Ideally you should start with a message here, just in case the test fails without any assertion failed. 
```suggestion
	lastErr := errors.New("no alerts asserted")
```

Reply:
Done